pkg/scheduler/backend/queue/scheduling_queue.go

```diff
@@ -878,49 +878,56 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 
+	start := time.Now()
+	events := []framework.ClusterEvent{}
+	if p.isSchedulingQueueHintEnabled {
+		// When unscheduled Pods are updated, we check with QueueingHint
+		// whether the update may make the pods schedulable.
+		// Plugins have to implement a QueueingHint for Pod/Update event
+		// if the rejection from them could be resolved by updating unscheduled Pods itself.
+		events = framework.PodSchedulingPropertiesChange(newPod, oldPod, true)
+	}
+
 	if p.isSchedulingQueueHintEnabled {
 		// The inflight pod will be requeued using the latest version from the informer cache, which matches what the event delivers.
 		// Record this update as Pod/Update because
 		// this update may make the Pod schedulable in case it gets rejected and comes back to the queue.
 		// We can clean it up once we change updatePodInSchedulingQueue to call MoveAllToActiveOrBackoffQueue.
 		// See https://github.com/kubernetes/kubernetes/pull/125578#discussion_r1648338033 for more context.
-		if exists := p.activeQ.addEventIfPodInFlight(oldPod, newPod, framework.UnscheduledPodUpdate); exists {
+		if exists := p.activeQ.addEventsIfPodInFlight(oldPod, newPod, events); exists {
 			logger.V(6).Info("The pod doesn't be queued for now because it's being scheduled and will be queued back if necessary", "pod", klog.KObj(newPod))
 			return
 		}
 	}
 
-	if oldPod != nil {
-		oldPodInfo := newQueuedPodInfoForLookup(oldPod)
-		// If the pod is already in the active queue, just update it there.
-		if pInfo := p.activeQ.update(newPod, oldPodInfo); pInfo != nil {
-			p.UpdateNominatedPod(logger, oldPod, pInfo.PodInfo)
-			return
-		}
+	oldPodInfo := newQueuedPodInfoForLookup(oldPod)
+	// If the pod is already in the active queue, just update it there.
+	if pInfo := p.activeQ.update(newPod, oldPodInfo); pInfo != nil {
+		p.UpdateNominatedPod(logger, oldPod, pInfo.PodInfo)
+		return
+	}
 
-		// If the pod is in the backoff queue, update it there.
-		if pInfo, exists := p.podBackoffQ.Get(oldPodInfo); exists {
-			_ = pInfo.Update(newPod)
-			p.UpdateNominatedPod(logger, oldPod, pInfo.PodInfo)
-			p.podBackoffQ.AddOrUpdate(pInfo)
-			return
-		}
+	// If the pod is in the backoff queue, update it there.
+	if pInfo, exists := p.podBackoffQ.Get(oldPodInfo); exists {
+		_ = pInfo.Update(newPod)
+		p.UpdateNominatedPod(logger, oldPod, pInfo.PodInfo)
+		p.podBackoffQ.AddOrUpdate(pInfo)
+		return
 	}
 
 	// If the pod is in the unschedulable queue, updating it may make it schedulable.
 	if pInfo := p.unschedulablePods.get(newPod); pInfo != nil {
```

Comment:
Can we remove `if pInfo := p.unschedulablePods.get(newPod); pInfo != nil {...}` and also remove preCheck of pkg/scheduler/eventhandlers.go#L196?
So, meaning, we let `MoveAllToActiveOrBackoffQueue` move the unscheduled Pod itself too.

Reply:
> So, meaning, we let MoveAllToActiveOrBackoffQueue move the unscheduled Pod itself too.

We can't do this as we need to use `PodItself/Update` to requeue the unscheduled Pod itself, and use `unscheduledPod/Update` to requeue other unscheduled pods.

Thus we need to update and requeue the unscheduled Pod itself, and then trigger a `MoveAllToActiveOrBackoffQueue` to move other pods.